src/consumer/options.rs

```diff
@@ -76,4 +78,10 @@ impl ConsumerOptions {
         self.initial_position = initial_position;
         self
     }
+
+    #[cfg_attr(feature = "telemetry", tracing::instrument(skip_all))]
+    pub fn with_receiver_queue_size(mut self, size: u32) -> Self {
+        self.receiver_queue_size = Some(if size == 0 { 1000 } else { size });
```

Comment:
Could you add a test for `size == 0` case as well as a TODO for handling this case? It would also be better opening an issue